core/src/main/scala/org/apache/spark/AccumulatorV2.scala

```diff
@@ -21,33 +21,77 @@ import java.{lang => jl}
 import java.io.ObjectInputStream
 import java.util.concurrent.ConcurrentHashMap
 import java.util.concurrent.atomic.AtomicLong
+import javax.annotation.concurrent.GuardedBy
+
+import scala.collection.mutable
 
 import org.apache.spark.scheduler.AccumulableInfo
 import org.apache.spark.util.Utils
 
-
+/**
+ * Metadata for the Accumulator
+ *
+ *
+ * @param countFailedValues whether to accumulate values from failed tasks. This is set to true
+ *                          for system and time metrics like serialization time or bytes spilled,
+ *                          and false for things with absolute values like number of input rows.
+ *                          This should be used for internal metrics only.
+
+ * @param dataProperty Data property accumulators will only have values added once for each
```

Comment:
I think `dataProperty` might possibly be clearer than `countOnRetry`, what do @squito / @rxin think?